crates/fsrt/src/main.rs

```diff
@@ -184,33 +211,103 @@ fn collect_sourcefiles<P: AsRef<Path>>(root: P) -> impl Iterator<Item = PathBuf>
         })
 }
 
-#[tracing::instrument(level = "debug")]
-fn scan_directory(dir: PathBuf, function: Option<&str>, opts: &Args) -> Result<()> {
-    let mut manifest_file = dir.clone();
-    manifest_file.push("manifest.yaml");
-    if !manifest_file.exists() {
-        manifest_file.set_extension("yml");
-    }
-    debug!(?manifest_file);
+pub fn scan_directory_test(forge_test_proj: Vec<MockForgeProject>) -> Option<Vec<Vulnerability>> {
+    scan_directory(PathBuf::new(), &Args::parse(), true, forge_test_proj)
+}
 
-    let manifest = fs::read_to_string(&manifest_file).into_diagnostic()?;
-    let manifest: ForgeManifest<'_> = serde_yaml::from_str(&manifest).into_diagnostic()?;
-    let name = manifest.app.name.unwrap_or_default();
+#[tracing::instrument(level = "debug")]
+fn scan_directory<'a>(
+    dir: PathBuf,
+    opts: &Args,
+    test: bool,
+    forge_test_proj: Vec<MockForgeProject>,
+) -> Option<Vec<Vulnerability>> {
+    let manifest_text;
+    let manifest = if test {
+        let mut forge_manifest_test = ForgeManifest::default();
```

Comment:
It might be better to have the manifest be created when instantiating the `MockForgeProject`s for each test case. You can add helper methods to make this less verbose if you'd like as well, as the easier it is to write test cases, the more likely we'll do it.

Reply:
Oh, yes. This sounds good. I was planning to do this in a different PR. I am trying to cut down on my diff size for faster turn around. But, yeah, I'd like the ability to add different files as well. I can change this 🙂